### ISSUE TITLE
How to add @apply rules for a class, but have base styles override them?

### ISSUE THREAD

**Issue by tnorthcutt** [nuxt-community/tailwindcss-module#120]
_tl;dr Codesandbox reproduction: https://codesandbox.io/s/nuxt-tailwindcss-o53yl_

I'm attempting to add some css rules that use `@apply` for elements within an element that has a `.markdown` class, similar to the example given in the Tailwind docs: https://tailwindcss.com/docs/adding-base-styles/#using-css

Right now, this works, but the custom css isn't overridden by the regular Tailwind utilities. 

My `tailwind.css` file looks like:

```
@import "tailwindcss/base";
@import "tailwindcss/components";
@import "tailwindcss/utilities";

.markdown h2 {
  @apply text-2xl text-gray-600;
}
```

The following html in a component results in the first `h2` **not** having a size of `text-4xl` or a color of `text-gray-600`:

```
<div class="markdown">
  <h2 class="text-4xl text-gray-900">This text should be larger</h2>
</div>
<h2 class="text-4xl text-gray-900">This text is larger</h2>
```

The Tailwind docs say "Define any of your own custom base styles directly after @tailwind base and before @tailwind components to avoid specificity issues"

However, if I move my custom base styles in my css file, I get the following compile error: `@import must precede all other statements (besides @charset)`

**Here's a reproduction code sandbox: https://codesandbox.io/s/nuxt-tailwindcss-o53yl**

Is there a way to structure things such that the `.markdown` rules work, but adding e.g. `.text-4xl` to an element overrides them?

**Comment by ossipov** [comment]
You've added a class to `h2` tag inside index.vue, but it got overwritten by a more specific rule `.markdown h2`. You can create a <style> inside of your index.vue and overwrite `.markdown h2`
```
<style>
.markdown h2 {
  @apply text-4xl text-gray-900;
}
</style>
```

**Comment by tnorthcutt** [comment]
@ossipov Thanks, that's helpful as a workaround.

I am still wondering if there is a way to structure my configuration such that in-html classnames on individual elements override my "extra" markdown `@apply` rules.
